### PR TITLE
skewt_1

### LINKED ISSUE #132: skewt_1
Description: Examples of various skew-t plot styles

Script URL: https://www.ncl.ucar.edu/Applications/Scripts/skewt_1.ncl
Plot URL: https://www.ncl.ucar.edu/Applications/Images/skewt_1_1_lg.png
https://www.ncl.ucar.edu/Applications/Images/skewt_1_2_lg.png
https://www.ncl.ucar.edu/Applications/Images/skewt_1_3_lg.png

Category: Skew-T

### PR DESCRIPTION
**Pull request by hCraker**
Closes #132 
The original NCL example creates three plots. The first a skew-t plot using NCL's defaults. The second adds shading between every other pair of isotherms and a height scale, and the third is the same as the second except it uses metric units.

Due to limitations in Python and with MetPy, I decided to make two figures. The first shows MetPy's default style for a skew-t. The second is a close approximation to the NCL default styling with the isotherm shading.

MetPy forces the pressure scale to be in hPa and the temperature to be in Celsius regardless of the units attached to the plotted data. Because of this, there isn't a good way for me to make an equivalent plot using customary units. Additionally, the axes labels are missing. This is because MetPy automatically adds the labels after data is plotted. I'm not sure how to convert height in meters to pressure level, so I have omitted the altitude scale for now.

### CODE REVIEW

**Comment by hCraker** [comment]
image
image

File: Plots/Skew-T/NCL_skewt_1.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+skew.plot_moist_adiabats()
+skew.plot_mixing_lines()
+
+gvutil.set_titles_and_labels(ax, maintitle="MetPy Default Skew-T")
+plt.show()
+plt.close()
+
+###############################################################################
+# Plot Skew-T that is similar to NCL's default Skew-T plot
+## Note that MetPy forces the x axis scale to be in Celcius and the y axis
```
Comment (anissa111) [inline Plots/Skew-T/NCL_skewt_1.py:41]:
```suggestion
## Note that MetPy forces the x axis scale to be in Celsius and the y axis
```

File: Plots/Skew-T/NCL_skewt_1.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+skew.plot_dry_adiabats()
+skew.plot_moist_adiabats()
+skew.plot_mixing_lines()
+
+gvutil.set_titles_and_labels(ax, maintitle="MetPy Default Skew-T")
+plt.show()
+plt.close()
+
+###############################################################################
+# Plot Skew-T that is similar to NCL's default Skew-T plot
```
Comment (anissa111) [inline Plots/Skew-T/NCL_skewt_1.py:40]:
```suggestion
# Plot Skew-T that is similar to NCL's default Skew-T plot

```
Adding spacing between the top line and secondary comment lines for better sphinx generation

File: Plots/Skew-T/NCL_skewt_1.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import matplotlib.pyplot as plt
+import matplotlib.lines as mlines
+import numpy as np
+from metpy.plots import SkewT
+from metpy.units import units
+
+import geocat.viz.util as gvutil
+
+###############################################################################
+# Plot Skew-T with MetPy Defaults:
```
Comment (anissa111) [inline Plots/Skew-T/NCL_skewt_1.py:23]:
```suggestion
# Plot Skew-T with MetPy Defaults:

```
Adding spacing between the top line and secondary comment lines for better sphinx generation

File: Plots/Skew-T/README.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+.. _skewt_examples:
+
+.. skewt-examples-index:
+
+Skew-T
+===
```
Comment (anissa111) [inline Plots/Skew-T/README.rst:6]:
```suggestion
======
```
Just extending the underline through the full title

**Comment by jukent** [comment]
This is great! Could you add a 3rd plot that adds some temperature, dew point temperature, and pressure data? Perhaps shading in CAPE too?

**Comment by hCraker** [comment]
> This is great! Could you add a 3rd plot that adds some temperature, dew point temperature, and pressure data? Perhaps shading in CAPE too?

The NCL example gallery plots data in a separate example [skewt_2](https://www.ncl.ucar.edu/Applications/Images/skewt_2_2_lg.png). I could certainly make this current comprehensive example script showing plotting data and shading CAPE if you think that would be better.

**Comment by hCraker** [comment]
I believe I fixed the issue with the first figure not showing up. Please let me know if the documentation still needs work. @anissa111

**Comment by hCraker** [comment]
I have fixed the merge conflict that was causing the build checks to stall. You all should be good to review this now. @anissa111 @michaelavs @jukent

**Review by michaelavs — APPROVED** [review]
Great job with this!

**Comment by hCraker** [comment]
> > This is great! Could you add a 3rd plot that adds some temperature, dew point temperature, and pressure data? Perhaps shading in CAPE too?
> 
> The NCL example gallery plots data in a separate example [skewt_2](https://www.ncl.ucar.edu/Applications/Images/skewt_2_2_lg.png). I could certainly make this current comprehensive example script showing plotting data and shading CAPE if you think that would be better.

@jukent Did you want me to make a single comprehensive example or save adding data and shading CAPE for a separate example?

**Comment by jukent** [comment]
> > > This is great! Could you add a 3rd plot that adds some temperature, dew point temperature, and pressure data? Perhaps shading in CAPE too?
> > 
> > 
> > The NCL example gallery plots data in a separate example [skewt_2](https://www.ncl.ucar.edu/Applications/Images/skewt_2_2_lg.png). I could certainly make this current comprehensive example script showing plotting data and shading CAPE if you think that would be better.
> 
> @jukent Did you want me to make a single comprehensive example or save adding data and shading CAPE for a separate example?

2 examples is fine! Great work.

**Review by jukent — APPROVED** [review]

**Review by anissa111 — APPROVED** [review]
This looks really good Rose!

Before I go ahead and merge this example, I wanted to make a comment that for the moment, forked branches are not actually included in the checks, and so the documentation generation needs to be verified on forked branches. There may be a better way to accomplish this, but I created an identical, non-forked branch to run the RTD generation on before approving this merge.

**Comment by hCraker** [comment]
> This looks really good Rose!
> 
> Before I go ahead and merge this example, I wanted to make a comment that for the moment, forked branches are not actually included in the checks, and so the documentation generation needs to be verified on forked branches. There may be a better way to accomplish this, but I created an identical, non-forked branch to run the RTD generation on before approving this merge.

When I started working on this project in May, I was told to fork the repository and do my development there. Is this no longer the case then? Should I make new branches directly from `NCAR:master`?

**Comment by anissa111** [comment]
> @anissa111 When I started working on this project in May, I was told to fork the repository and do my development there. Is this no longer the case then? Should I make new branches directly from `NCAR:master`?

No, you're doing everything right! The way we have the github checks set up right now isn't correctly checking forked branches, but we obviously need to be able to have development forked branches, so for the time being, I just wanted to note before I merged that I did run the checks on this branch directly from readthedocs.

**Pull request merged**